src/integration-tests/src/firestore.rs

```diff
@@ -0,0 +1,162 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use firestore::Error;
+use firestore::Result;
+use firestore::client::Firestore;
+use firestore::model;
+use rand::{Rng, distr::Alphanumeric};
+
+pub const COLLECTION_ID_LENGTH: usize = 32;
+
+fn new_collection_id() -> String {
+    let prefix = "doc-";
+    let collection_id: String = rand::rng()
+        .sample_iter(&Alphanumeric)
+        .take(COLLECTION_ID_LENGTH - prefix.len())
+        .map(char::from)
+        .collect();
+    format!("{prefix}{collection_id}")
+}
+
+pub async fn basic(config: Option<gax::options::ClientConfig>) -> Result<()> {
+    // Enable a basic subscriber. Useful to troubleshoot problems and visually
+    // verify tracing is doing something.
+    #[cfg(feature = "log-integration-tests")]
+    let _guard = {
+        use tracing_subscriber::fmt::format::FmtSpan;
+        let subscriber = tracing_subscriber::fmt()
+            .with_level(true)
+            .with_thread_ids(true)
+            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+            .finish();
+
+        tracing::subscriber::set_default(subscriber)
+    };
+
+    cleanup_stale_documents().await?;
+
+    let project_id = crate::project_id()?;
+    let client = if let Some(config) = config {
+        Firestore::new_with_config(config).await?
+    } else {
+        Firestore::new().await?
+    };
+    let collection_id = new_collection_id();
+    let response = client
+        .create_document(
+            format!("projects/{project_id}/databases/(default)/documents"),
+            &collection_id,
+        )
+        .set_document(model::Document::new().set_fields([
+            (
+                "greeting",
+                model::Value::new().set_string_value("Hello World!"),
+            ),
+            (
+                "integration-test",
+                model::Value::new().set_boolean_value(true),
+            ),
+        ]))
+        .send()
+        .await?;
+    println!("SUCCESS on create_document: {response:?}");
+
+    let document_name = response.name;
+    let response = client.get_document(&document_name).send().await?;
+    println!("SUCCESS on get_document: {response:?}");
+
+    let mut documents = client
+        .list_documents(
+            format!("projects/{project_id}/databases/(default)/documents"),
+            &collection_id,
+        )
+        .paginator()
+        .await
+        .items();
+    while let Some(doc) = documents.next().await {
+        let doc = doc?;
+        println!("  ITEM = {doc:?}");
+    }
+    println!("SUCCESS on list_documents: {response:?}");
+
+    let response = client
+        .update_document(
+            model::Document::new()
+                .set_name(&document_name)
+                .set_fields([("greeting", model::Value::new().set_string_value("Goodbye."))]),
+        )
+        .set_update_mask(model::DocumentMask::new().set_field_paths(["greeting"]))
```

Comment:
comment: it would be cool if the GAPIC computed a default update mask for the application.

and `.set_update_mask(...)` can be used to override it if necessary (hopefully not).

https://google.aip.dev/134